pytensor/tensor/rewriting/basic.py

```diff
@@ -975,6 +976,30 @@ def local_join_make_vector(fgraph, node):
         return [ret]
 
 
+@register_specialize
+@register_canonicalize
+@register_useless
+@node_rewriter([Sum])
+def local_sum_make_vector(fgraph, node):
+    """A sum of a MakeVector node is just the sum of the elements."""
+    (array,) = node.inputs
+
+    if array.owner is None:
+        return
+
+    if not isinstance(array.owner.op, MakeVector):
+        return
+
+    if node.op.axis not in [None, 0, -1]:
```

Comment:
MakeVector doesn't have axis?

Reply:
This should be the axis of the Sum. I guess since we know that the inputs of the sum is 1d, we could also ditch that check?

Reply:
Yeah I updated my comment in the meantime

Reply:
Actually, it would be that the sum has axis == (), ie it is a no-op.
I added an extra case for that where we just remove the sum.

Reply:
I am surprised an empty axis is allowed